src/cargo/core/compiler/job_queue.rs

```diff
@@ -1239,7 +1249,16 @@ impl<'cfg> DrainState<'cfg> {
             if let FixableWarnings::Positive(fixable) = count.fixable {
                 // `cargo fix` doesnt have an option for custom builds
                 if !unit.target.is_custom_build() {
-                    let mut command = {
+                    // To make sure the correct command is shown for `clippy` we
+                    // check if `RUSTC_WORKSPACE_WRAPPER` is set. `clippy` sets
+                    // this when being ran so it can partially be identified
+                    // when this is set.
+                    let command = if rustc_workspace_wrapper.is_some() {
```

Comment:
Unfortunately clippy isn't the only tool which sets `RUSTC_WORKSPACE_WRAPPER`. I think some other approach will be needed here.

Reply:
I was worried about that, I was hoping it would be fine since [it was used before](https://github.com/rust-lang/cargo/blob/65535890dbaa2b186633aedc11e550c3990fc5e9/src/cargo/core/compiler/context/compilation_files.rs#L607). The only thing I can think of is using [`CLIPPY_ARGS`](https://github.com/rust-lang/rust-clippy/blob/51ec465cc3dfb62a2dad7e808b399fa76a1c9170/src/main.rs#L141), which appears to pass the test I wrote. I will update it to that.

Reply:
After some discussion, it was decided that `RUSTC_WORKSPACE_WRAPPER` would be better as it is already agreed upon by `clippy` and `cargo`. Since other things can use `RUSTC_WORKSPACE_WRAPPER`, we look for a wrapper named [`clippy-driver`](https://github.com/rust-lang/rust-clippy/blob/51ec465cc3dfb62a2dad7e808b399fa76a1c9170/src/main.rs#L120).